Show artifact versions

ffjlabo opened the pull request:
**What this PR does / why we need it**:

This PR changes to show artifact versions with links.

<img width="947" alt="PipeCD" src="https://user-images.githubusercontent.com/40124947/157580528-39feb59d-c96f-467f-a2c6-7d7b74380634.png">

<img width="1300" alt="PipeCD" src="https://user-images.githubusercontent.com/40124947/157580604-b2b1de14-6319-4407-9997-94780c8e012c.png">


**Which issue(s) this PR fixes**:

A part of https://github.com/pipe-cd/pipecd/issues/3303

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
-->
```release-note
Be able to show artifact versions with links
```

khanhtc1202 commented on web/src/components/application-detail-page/application-detail/index.tsx:
```diff
@@ -125,7 +125,30 @@ const MostRecentlySuccessfulDeployment: FC<{
+                    <>
+                      {i > 0 && ","}
+                      <Link
+                        href={`//${v.url}`}
```
should only add this `//` in case the url does not contain schema.
Can use v.url.includes(`://`) as the condition.

nghialv commented:
Love this!

What do you guys thing about changing the layout of the version?
For example, `v0.6.0 (helloworld)` -> `helloworld:v0.6.0`.

nghialv commented:
Btw, should we trim the version in case it is too long?

Like commit hash, we should show only the first 7 characters.

`helloworld:36515f2...` instead of `helloworld:sha36515f2d60ad86cc1f7b7e8d97e935741e5cf9f1`

ffjlabo commented:
> Love this!
> 
> What do you guys thing about changing the layout of the version? For example, `v0.6.0 (helloworld)` -> `helloworld:v0.6.0`.

I see, may be `helloworld:v0.6.0` style is natural for users because they often show that style in artifact files (e.g. k8s manifest). 



> Btw, should we trim the version in case it is too long?
> 
> Like commit hash, we should show only the first 7 characters.
> 
> `helloworld:36515f2...` instead of `helloworld:sha36515f2d60ad86cc1f7b7e8d97e935741e5cf9f1`

I agree with it!

ffjlabo commented:
How about the above?

nghialv commented:
> How about the above?

@khanhtc1202 @knanao what do you think?

khanhtc1202 commented:
yeh, one vote from me. Please go ahead and address it 🙏

ffjlabo commented:
These are after fixed.

<img width="786" alt="PipeCD" src="https://user-images.githubusercontent.com/40124947/157614379-c77d7792-e3fe-4af7-b56d-e45f752a81a9.png">

<img width="1409" alt="PipeCD" src="https://user-images.githubusercontent.com/40124947/157614270-9bfb1faf-e166-4d24-91c4-95bceb2a8e17.png">

khanhtc1202 commented:
nits, I think we should only trim the long text as it's a table cell content. On the detail page, we should just show it clearly, wdyt?

ffjlabo commented:
I see! Then I'll fix to truncate it in case the long text over the size of the table cell.

ffjlabo commented:
On the detail page, I fixed it to show full version info and split them into multiple lines.
<img width="1140" alt="PipeCD" src="https://user-images.githubusercontent.com/40124947/157625320-f19f2544-9dcb-45df-bf2e-46fdd4dc69e8.png">

nghialv commented:
Looks beautiful! ❤️ 
/lgmt

ffjlabo commented:
sorry I found some mistakes, so I'll fix

ffjlabo commented:
I fixed, PTAL. 🙇

nghialv commented:
👍 
/lgtm

khanhtc1202 commented:
it's beautiful, thank you 😀 
/approve
